src/Executor.cc

```diff
@@ -123,7 +123,7 @@ void Executor::run_asynchronously(Sequence& sequence, Context context)
                          std::move(context), comm_channel_);
 
     sequence.set_running(true);
-    sequence.set_error_message("");
+    sequence.set_error(gul14::nullopt);
```

Comment:
Maybe easier to read would be

```c++
void set_error(gul14::optional<Error> opt_error = {});

...

sequence.set_error();
```

But then, maybe not. Maybe I just dislike the explicit `nullopt`, and in principe you can also

```c++
sequence.set_error({ }); // set no error, obviously
```

Reply:
I am not sure that `set_error()` without an argument is really expressive. I think I like the empty braces, but I won't change any code now.